atomicfu-transformer/src/main/kotlin/kotlinx/atomicfu/transformer/AtomicFUTransformer.kt

```diff
@@ -600,6 +659,7 @@ class AtomicFUTransformer(
                             val next = i.nextUseful
                             val fieldId = (next as? FieldInsnNode)?.checkPutField()
                                 ?: abort("factory $methodId invocation must be followed by putfield")
+                            eraseTraceInit(i.previous)
```

Comment:
It does not work. It attempts to `eraseTraceInit` even if there is no trace parameters in this constructor invocation. It fails on this code:

```
class Counter {
    private val a = atomic(0)
    private val trace = trace(64)

    fun inc(): Int {
        trace { "inc() invoked" }
        val x = a.incrementAndGet()
        trace { "inc() = $x" }
        return x
    }

    internal fun get() = a.value
}
``` 

with

```
Trace.kt: kotlinx.coroutines.experimental.channels.Trace::<init>: Failed to transform: java.lang.ClassCastException: org.objectweb.asm.tree.InsnNode cannot be cast to org.objectweb.asm.tree.MethodInsnNode
java.lang.ClassCastException: org.objectweb.asm.tree.InsnNode cannot be cast to org.objectweb.asm.tree.MethodInsnNode
        at kotlinx.atomicfu.transformer.FlowAnalyzer.getInitStart(FlowAnalyzer.kt:64)
        at kotlinx.atomicfu.transformer.AtomicFUTransformer$TransformerMV.eraseTraceInit(AtomicFUTransformer.kt:619)
        at kotlinx.atomicfu.transformer.AtomicFUTransformer$TransformerMV.transform(AtomicFUTransformer.kt:662)
        at kotlinx.atomicfu.transformer.AtomicFUTransformer$TransformerMV.visitEnd(AtomicFUTransformer.kt:470)
```